app/views/creatures/_form.html.erb

```diff
@@ -0,0 +1,22 @@
+<%= form_with model: [creature.campaign, creature], html: { data: { action: "ajax:success->modal#close ajax:error->modal#onPostError" } } do |form| %>
```

Comment:
How would you feel about the `ModalController` setting up listeners for these actions instead?

Reply:
hm, I'm not quite sure how it would do that... would it need to know about the existence of the form in advance to set up the listener?

I could probably figure that out, but I like the idea that the form is explicitly setting up the behavior it wants and to me it's very clear, instead of having side effects elsewhere. As it is, I'm not crazy about the `onPostError` being on the modal controller. It has more to do with remote forms then the modal specifically. I actually thought about creating a separate form controller for it, but then putting the modal#close there could be weird. 

If you did that, then you'd end up with something like this on the form

```
{ data: { action: "ajax:success->modal#close ajax:error->form#replace" } }
```

I guess that's not too bad now that I type it out.

I'm inclined to move forward with this as is, and then we can revisit this "api" as I think we should essentially convert all forms to be remote and even inside of the modal (like new map).

Reply:
Yeah, not worth getting caught up in now. Definitely a space I'm interested in exploring more though!

I realised I was also working with one completely imagined constraint in my head that I'd like the HTML partials to be usable in both the HTML and JS layouts.